packages/features/ee/workflows/components/WorkflowStepContainer.tsx

```diff
@@ -356,20 +398,76 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
                   {isPhoneNumberNeeded && (
                     <>
                       <Label className="pt-4">{t("custom_phone_number")}</Label>
-                      <PhoneInput<FormValues>
-                        control={form.control}
-                        name={`steps.${step.stepNumber - 1}.sendTo`}
-                        placeholder={t("phone_number")}
-                        id={`steps.${step.stepNumber - 1}.sendTo`}
-                        className="w-full rounded-md"
-                        required
-                      />
+                      <div className="block sm:flex">
+                        <PhoneInput<FormValues>
+                          control={form.control}
+                          name={`steps.${step.stepNumber - 1}.sendTo`}
+                          placeholder={t("phone_number")}
+                          id={`steps.${step.stepNumber - 1}.sendTo`}
+                          className="min-w-fit sm:rounded-tl-md sm:rounded-bl-md sm:border-r-transparent"
+                          required
+                          onChange={() => {
+                            const isAlreadyVerified = !!verifiedNumbers
```

Comment:
Everytime the phone number input changes, we check if the phone number is already verified